rfcs/20230323-add-split-graph-op/README.md

```diff
@@ -0,0 +1,161 @@
+# Add Split Operation in Graph API
+
+## Background
+### Motivation
+
+`Split` operation splits an input tensor into pieces of the specific length
+along some axis, which is opposite to the operation of `concat`. `Split`
+operation is supported by different deep learning frameworks and toolkit, eg.
+ PyTorch [1], TensorFlow [2], OpenVINO [3], and ONNX [4]. The `split`+`concat` 
+ pattern is more common in most models and takes up a large amount of computing 
+ time. However, since the graph API does not support `Split` op, possible 
+ optimization is limited. So we propose to add the `Split` op into graph API.
+
+### Investigation
+
+The `Split` operation is supported by most frameworks. To split an input tensor, 
+we need to specify the split dimension according to the `axis` for input tensor, 
+and then output multiple small chunks. But as for how to split along specific 
+dimensions, the definition of frameworks is slightly different, which usually 
+includes three rules. 
+- Specify the number of output tensors through an integer
+- Uniformly specify the size of each output chunk through an integer
+- Specify the specific size of each chunk through a list. 
+
+The size of all output chunks in the first two rules is the same but may differ 
+in the third rule. Different frameworks support one or more of the above three 
+rules in different ways. For example, TensorFlow supports the first and third 
+rules through `num_or_size_splits`, while Pytorch supports the second and third 
+rules through `split_size_or_sections`. In addition, for different rules, the 
+definitions of different frameworks are also different, for example, some allow 
+divisibility while others do not. The following are the brief definitions of 
+`Split` in four common frameworks. For details, please refer to the link at the 
+end of this RFC.
+
+#### operations used in Frameworks and toolkit
+
+| Tensorflow | Argument Name         | Description                                                                                       |
```

Comment:
As TF workloads moves to XLA compiler path, and Pytorch move to Torchdynamic/TorchInductor, FW operations are decomposed to XLA HLO op or Pytorch PrimTorch OP.  These operations usually has simpler semantics.  oneDNN Graph operation should first consider what is the basic operation we need to support, and usually these operation design would give better insight - what operations are popular (those unpopular ones may not lowered to these simple ops), and how the basic operations look like. 

Could you please also include how these two set operations support split?

Reply:
Hi, @Jianhui-Li, just checked the XLA and PrimTorch, these two operations don't directly support `Split`. 
- TensorFlow's `Split` will be lowered to XLA HLO [Slice op](https://www.tensorflow.org/xla/operation_semantics#slice). I have updated the definition of XLA HLO `Slice` in this section.  
- For Pytorch/PrimTorch, when Torch IR is lowered to Aten IR/Prim IR, the `torch.Split` will call `aten.Split` and no decomposition to PrimTorch ops happened during this stage. And inductor will lower `aten.Split` to `slice` in its implementation.  cc @retonym @chunyuan-w